vcn_gateways.tf

```diff
@@ -23,8 +23,53 @@ resource "oci_core_route_table" "ig" {
   freeform_tags = var.tags
 
   route_rules {
+    # * With this route table, Internet Gateway is always declared as the default gateway
     destination       = local.anywhere
     network_entity_id = oci_core_internet_gateway.ig[0].id
+    description       = "Terraformed - Auto-generated at Internet Gateway creation: Internet Gateway as default gateway"
+  }
+
+  dynamic "route_rules" {
+    # * filter var.internet_gateway_route_rules for routes with "drg" as destination
+    # * and steer traffic to the module created DRG
+    for_each = var.internet_gateway_route_rules != null ? { for k, v in var.internet_gateway_route_rules : k => v
+    if v.network_entity_id == "drg" } : {}
+
+    content {
+      destination       = route_rules.value.destination
+      destination_type  = route_rules.value.destination_type
+      network_entity_id = oci_core_drg.drg[0].id
+      description       = route_rules.value.description
+    }
+  }
+
+  dynamic "route_rules" {
+    # * filter var.internet_gateway_route_rules for routes with "internet_gateway" as destination
```

Comment:
line 25-30 is the pre-existing code, that I wanted to keep out of this PR, with no behavior change as you are suggesting: if Internet Gateway creation is enabled, the module automatically creates a Route Table that **always** contains a rule for unknown traffic to the gateway (here internet gateway, same logic with nat gateway).

The change's intent is to give user the possibility to **add** custom rules.

Long term evolution, I think these automatically created route tables should be kept out of the vcn module. Route Table creation is something that needs to be taken care of at subnet creation time. I am baking an independent route table module for that.

Reply:
@slevey, @kral2 and I discussed this. we need that default route so that the default route rule can be created. We cannot pass it as default in variables as the internet gateway won't be created yet and therefore we won't know its id. same for nat gw below. I think LGTM now.